### DISCUSSION TITLE
Which optimization approach should be implemented?

### DISCUSSION THREAD

**Discussion by zsStrike** [futurewei-cloud/chogori-platform#145] [Q&A]
In the chapter 3.8.4 in `K23SI-Detailed-States` document, there are two approaches for general optimization:

+ the client keeps track of what write operation has already got persistence confirmed...
+ the TRH keeps track of what write operation has already got persistence confirmed...

I wonder if the two approaches should be both implemented or just implement the latter as we discussed in last few weeks?

**Comment by iafuture** [comment]
i believe this was answered in the meetings. The optimization would be to allow TRH to keep track of completed writes
